slotsmanager.c

```diff
@@ -406,25 +406,29 @@ slots_gc_sorted()
 }
 
 void
-slots_reserve(int atleast[6])
+slots_reserve(int atleast[6], int collective)
 {
     int newMaxSlots[6];
     int ptype;
     int good = 1;
+    /*Ensure all processors have initially the same number of particle slots*/
+    if(collective)
```

Comment:
It appears that other than the allreduce collective only affects the message output.. This will probably look awkward when we revisit it in a year or two.

Reply:
What do you suggest?

Reply:
I couldn't think of anything better. Do we really need uncollective growth?

Reply:
We do - some processors may be over-weight black holes and others may be over-weight stars. It is *collective* growth we don't really need.

Reply:
So we shall perhaps get rid of collective growth and wrap the message in DEBUG. for the special case where we explicitly need a collective growth then we do the allreduce before calling the function.

Reply:
How about do the allreduce before but keep an argument like "int where" which controls the message (same name as in message() and endrun())? Having the allocated message is fairly useful even if DEBUG is off (although we could reword it a little).